torch/_dynamo/eval_frame.py

```diff
@@ -79,7 +79,7 @@ def __getattr__(self, name):
         return getattr(self._orig_mod, name)
 
     def forward(self, *args, **kwargs):
-        return self.dynamo_ctx(self._orig_mod.forward)(*args, **kwargs)
+        return self.dynamo_ctx(self._orig_mod.__call__)(*args, **kwargs)
```

Comment:
This change looks innocuous enough.  But it's actually concerning me a bit.

We're asymmetrically mapping from `forward` of the outer 'OptimizedModule' object to `__call__` of the wrapped module.  This is probably not a good idea.

I would feel better if we could just change OptimizedModule implementation so it does not implement forward at all, and instead implements `__call__`.  Would this be OK though?  And does this even matter or is it just a nit?

Reply:
Are you using __call__ here to force hitting the hooks?

Reply:
right. referring to my notes in the Commit Message / PR Desc, This is the case (2) where hooks were applied to the outermost module m passed to compile(m), and would still be missed if we only support `__call__` inside NNModuleVariable.

And the reason changing `forward` to `__call__` helps with hooks is that torch NNModule implements __call__ to call hooks and then call forward, more or less.

Reply:
> I would feel better if we could just change OptimizedModule implementation so it does not implement forward at all, and instead implements __call__. Would this be OK though? And does this even matter or is it just a nit?

Ditto, wondering why we don't just change line 81 to `__call__`. Also, it seems fine in principle to keep the forward stub around, going to inner forward.